iceoryx_posh/include/iceoryx_posh/internal/popo/response_deleter.hpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2021 - 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_POSH_POPO_RESPONSE_DELETER_HPP
+#define IOX_POSH_POPO_RESPONSE_DELETER_HPP
+
+#include "iceoryx_posh/mepoo/chunk_header.hpp"
+#include "iceoryx_posh/popo/rpc_header.hpp"
+
+namespace iox
+{
+namespace popo
+{
+///
+/// @brief The ResponseDeleter struct is a custom deleter in functor form which releases loans to a response's
+/// underlying memory chunk via the corresponding port..
+/// Each port should create its own instance of this deleter struct.
+/// @tparam Port is either the ClientPortUser or ServerPortUser and need to have a `releaseResponse` method
+template <typename Port>
+struct ResponseDeleter
```

Comment:
Do we really need `Request/ResponseDeleter` now that we have a `cxx::function` at our disposal which could be used in the `cxx::unique_ptr`

Reply:
Good question. I really can't answer that without a closer look at the `SmartChunk` but I currently lack some time to do so. Maybe we could completely get rid of the deleter but I'm not sure about the effect on the `Sample` and and it's tests

Reply:
What do you think about adding this to the issue with using `cxx::function` in iceoryx. I could add a task to check whether we can get rid of `SampleDeleter`, `RequestDeleter` and `ResponseDeleter`

Reply:
I already created an issue: https://github.com/eclipse-iceoryx/iceoryx/issues/1143

I would leave it as it is so that we have nice and small issues.